pkg/store/bucket.go

```diff
@@ -1320,23 +1324,72 @@ func (r *bucketIndexReader) ExpandedPostings(ms []*labels.Matcher) ([]uint64, er
 			return nil, errors.Wrap(err, "toPostingGroup")
 		}
 
+		// Intersection with empty postings would return no postings anyway.
+		if pg.alwaysEmptyPostings() {
```

Comment:
`IsEmptyPostings` is a different guarantee... there can be labels that are not present in the block, and will also return empty postings. `IsEmptyPostings` name would suggest that it returns true in such case, but it doesn't. On the other hand, 'IsAlwaysEmptyPostings` (new suggested name) says that this will be empty no matter the block.

Reply:
Removed this method in favor of checking for specific `emptyPostingsGroup` sentinel value.

Reply:
And now remove check for sentinel value again, since it's incorrect. We really need to check if group is empty. There are other ways how group can be empty, e.g. by using `label="non-existant-value"` matcher.